activerecord/lib/active_record/reflection.rb

```diff
@@ -11,6 +11,7 @@ module Reflection # :nodoc:
       class_attribute :_reflections, instance_writer: false, default: {}
       class_attribute :aggregate_reflections, instance_writer: false, default: {}
       class_attribute :automatic_scope_inversing, instance_writer: false, default: false
+      class_attribute :automatically_invert_plural_associations, instance_writer: false, default: false
```

Comment:
@casperisfine https://github.com/rails/rails/pull/50280 also involves automatic inversion (for delegated type associations).

If it were to be merged, would it make sense to put it behind a configuration flag of its own? Would it be appropriate to make *this* PR's configuration flag general enough to also apply for both scenarios?

Reply:
Ideally anything that could cause backward compatibility issues should be behind a flag to allow to deprecation and selective upgrade, so yes. 

As for sharing the same flag, I think it would be weird because the names wouldn't match.